library/azure_rm_devtestlabvirtualmachine.py

```diff
@@ -0,0 +1,516 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Zim Kalinowski, <zikalino@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_devtestlabvirtualmachine
+version_added: "2.8"
+short_description: Manage Azure DevTest Lab Virtual Machine instance.
+description:
+    - Create, update and delete instance of Azure DevTest Lab Virtual Machine.
+
+options:
+    resource_group:
+        description:
+            - The name of the resource group.
+        required: True
+    lab_name:
+        description:
+            - The name of the lab.
+        required: True
+    name:
+        description:
+            - The name of the virtual machine.
+        required: True
+    notes:
+        description:
+            - The notes of the virtual machine.
+    os_type:
+        description:
+            - The OS type of the virtual machine.
+    size:
+        description:
+            - The size of the virtual machine.
+    user_name:
+        description:
+            - The user name of the virtual machine.
+    password:
+        description:
+            - The password of the virtual machine administrator.
+    ssh_key:
+        description:
+            - The SSH key of the virtual machine administrator.
+    lab_subnet_name:
```

Comment:
should support resource id format, dict of resource group  + name, and name

Reply:
i was wondering about that, and finally decided not to, because that vm and subnet must be in the same lab, so there's no point to support entire id.
perhaps we should have single "lab_subnet" parameter and that could either be a string and include entire id, or **name** / **virtual_network_name**.
**resource_group** and **lab_name** are known.